core/src/appleMain/kotlin/CentralManager.kt

```diff
@@ -6,25 +6,42 @@ import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.withContext
 import platform.CoreBluetooth.CBCentralManager
+import platform.CoreBluetooth.CBCentralManagerOptionRestoreIdentifierKey
 import platform.CoreBluetooth.CBCharacteristic
 import platform.CoreBluetooth.CBCharacteristicWriteType
 import platform.CoreBluetooth.CBDescriptor
 import platform.CoreBluetooth.CBPeripheral
 import platform.CoreBluetooth.CBService
 import platform.CoreBluetooth.CBUUID
 import platform.Foundation.NSData
+import platform.Foundation.NSUUID
+import platform.Foundation.NSUserDefaults
 
 private const val DISPATCH_QUEUE_LABEL = "central"
+private const val CBCENTRALMANAGER_RESTORATION_ID = "kable-central-manager"
+private const val CBCENTRALMANAGER_CONSUMER_ID_KEY = "kable-central-manager-consumer-id"
 
 public class CentralManager internal constructor() {
 
     internal companion object {
         val Default: CentralManager by lazy { CentralManager() }
     }
 
+    private val userDefaults = NSUserDefaults.standardUserDefaults
+
+    // This value is needed to ensure multiple instances of Kable running on the same iOS device
+    // do not cross pollinate restored instances of CBCentralManager. The value will live for the
+    // lifetime of the consuming app.
+    private val consumerId: String
+        get() = userDefaults.stringForKey(CBCENTRALMANAGER_CONSUMER_ID_KEY)
+            ?: NSUUID().UUIDString().also {
+                userDefaults.setObject(it, CBCENTRALMANAGER_CONSUMER_ID_KEY)
+            }
+
     private val dispatcher = QueueDispatcher(DISPATCH_QUEUE_LABEL)
     internal val delegate = CentralManagerDelegate()
-    private val cbCentralManager = CBCentralManager(delegate, dispatcher.dispatchQueue)
+    private val cbOptions = mutableMapOf<Any?, Any>(CBCentralManagerOptionRestoreIdentifierKey to CBCENTRALMANAGER_RESTORATION_ID + consumerId).toMap()
```

Comment:
The way this is concatenated it will make for some ugly values, e.g.: `kable-central-managerd97f4db7-2bbf-424b-99f2-a0dc224d086a`

The concatenation should probably read: `"$CBCENTRALMANAGER_RESTORATION_ID-$consumerId"`

Reply:
I can change it if you really want, but this value is not meant to be seen by any humans. It shouldn't appear in any logging or queue names; it's strictly just for the OS to be able to tell two instances apart in case Kable happens to be running in another iOS app.